client/profile-wizard/steps/product-types/test/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import { setSetting } from '@woocommerce/wc-admin-settings';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import ProductTypes from '../';
+
+describe( 'ProductTypes', () => {
+	beforeEach( () => {
+		setSetting( 'onboarding', {
+			productTypes: {
+				paidProduct: {
+					description: 'Paid product type',
+					label: 'Paid product',
+					more_url: 'https://woocommerce.com/paid-product',
+					product: 100,
+					slug: 'paid-product',
+					yearly_price: 120,
+				},
+				freeProduct: {
+					label: 'Free product',
+				},
+			},
+		} );
+	} );
+
+	afterEach( () => {
+		setSetting( 'onboarding', {} );
+	} );
+
+	test( 'should render product types', () => {
+		const { container } = render( <ProductTypes /> );
+		expect( container ).toMatchSnapshot();
+	} );
+
+	test( 'should show annual prices on toggle', () => {
+		const { container } = render( <ProductTypes /> );
+
+		const toggle = screen.getByLabelText( 'Display monthly prices', {
+			selector: 'input',
+		} );
+
+		userEvent.click( toggle );
+
+		expect( container ).toMatchSnapshot();
+	} );
+
+	test( 'should show validation error on continue', async () => {
+		const mockGoToNextStep = jest.fn();
+		const mockUpdateProfileItems = jest.fn().mockResolvedValue();
+
+		render(
+			<ProductTypes
+				goToNextStep={ mockGoToNextStep }
+				updateProfileItems={ mockUpdateProfileItems }
```

Comment:
This does not seem to be mocking properly.  Can someone on @woocommerce/isotope take a look at this?

Reply:
Its a timing issue. If you remove the two `await` declarations from `onContinue` class method on `ProductTypes` it passes.

Hopefully, that leads you down the right path. Also, looking at the code, it appears `validateField` does not need to be an async method.

Reply:
It does look like `validateField` does not need to be async and I handled some other clean-up work that seems to be related to wc-api in here, but unfortunately that does not get the tests passing for me.

If you log the mocked functions, it does not look like they are being passed at all to the component.  Even mocking `@wordpress/data` to override `dispatch` does not seem to override anything.  The real `updateProfileItems` gets called each time.

Reply:
> The real `updateProfileItems` gets called each time.

The component under test is wired to the stores. You can add an additional export in this file that is the component class before it gets wrapped in `compose()` to aid in testing.

See the PayPal payment task for an example:
https://github.com/woocommerce/woocommerce-admin/blob/aac0bde3caa0d94d7404e2e6a8918af3dd387ddb/client/task-list/tasks/payments/paypal.js#L21
https://github.com/woocommerce/woocommerce-admin/blob/aac0bde3caa0d94d7404e2e6a8918af3dd387ddb/client/task-list/test/payments.js#L12

Reply:
Thanks, Jeff!